vertx-mysql-client/src/main/java/io/vertx/mysqlclient/MySQLConnectOptions.java

```diff
@@ -544,4 +547,16 @@ public JsonObject toJson() {
     MySQLConnectOptionsConverter.toJson(this, json);
     return json;
   }
+
+  @GenIgnore
+  @Override
+  public SocketAddress getSocketAddress() {
+    String path = getProperties().get("socket");
+    return path != null ? SocketAddress.domainSocketAddress(path) : super.getSocketAddress();
+  }
+
+  @GenIgnore
+  public boolean isUsingDomainSocket() {
+    return this.getProperties().containsKey("socket");
```

Comment:
Are we going this way which is different from the usage of postgres client(using host+port)?
The `socket` is MySQL specific property, how about keep representing the unix socket path with `host` field but add additional support for automatically converting `socket` to `host` in URL parser?

Reply:
`socket` is indeed MySQL specific. It is described in our documentation as supported, even if it actually isn't. Perhaps we should support both

Reply:
Done